test/e2e/scenarios/add_cluster.go

```diff
@@ -0,0 +1,18 @@
+package scenarios
+
+import (
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = PDescribe("Adding existng network policy to newly added cluster", func() {
+	PIt("Should implement existng network policy in newly added cluster", func() {
+		By("creating pod1 with label1 in cluster1 in namespace1")
+		By("creating pod2 with label2 in cluster2 in namespace1")
+		By("creating network policy with namespace selector")
+		By("testing connectivity between pods")
+		By("Create a new cluster")
```

Comment:
More than "create a cluster, discover the cluster"

could be "Registering cluster3 to kubefed"


We need to unregister such cluster after the test ends.



Creating them dynamically seems unfeasible (doable with kind, but think of running this test with real clusters..)

Reply:
We can run add_cluster.go and then remove_cluster.go in series so that way unregistering cluster 3 will be part of remove_cluster.go/ WDYT?